### ISSUE TITLE
Socket Problem in Windows Server

### ISSUE THREAD

**Issue by dladeli** [murataya/o2popper#6]
Hi i need support after installation of program in Windows Server 2019.
After complete the configuration e click in Auth... i receve this error:

winerror 10013 an attempt was made to access a socket in a way forbidden by its access permissions

The server dont have Antivirus e Windwos defender it's disable.
Firewall disable.....
The port used is 12345 and it's not used.

I dont have idea where is the problem.....

Can you help me?

![Screenshot 2023-03-02 155509](https://user-images.githubusercontent.com/130652611/231717753-f1af3e91-e64c-4ac0-a300-4aad82d9725d.png)

**Comment by murataya** [comment]
I'm sorry. I haven't tried it on Windows Server 2019.

The default redirect port for OAuth2 is 8080.
If you need to change it, add the following settings to your secret file.
"_redirect_port":12345

Is your secret file (client_secret_xxxx.js) set like this?

**Comment by dladeli** [comment]
Hi, thank's for reply, can you send an example? Because in client secret it's present a lot of { }.

{"installed":{"client_id":"","auth_uri":"https://login.microsoftonline.com/common/oauth2/v2.0/authorize","token_uri":"https://login.microsoftonline.com/common/oauth2/v2.0/token","client_secret":"","_scopes":["offline_access","https://outlook.office.com/POP.AccessAsUser.All","https://outlook.office.com/IMAP.AccessAsUser.All","https://outlook.office.com/SMTP.Send"],"_pop_server":"outlook.office365.com:995","_imap_server":"outlook.office365.com:993","_smtp_server":"smtp.office365.com:587","_redirect_port":12345}}

It's correct the my edit?

PS: im not insered the client id and the secrect code :-)

**Comment by murataya** [comment]
Your edit is correct.
It seems that the reason why it doesn't work is somewhere else.

**Comment by dladeli** [comment]
Ok it's work but, after browser it's open i see a blank page.
![aaaa](https://user-images.githubusercontent.com/130652611/231970325-57028626-efa2-4725-8922-a2fa697d806f.png)

**Comment by dladeli** [comment]
I hace chenged the browser it's work thank's a lot.

Another question....

It's possibile to reduce the time of occupation socket for multi send?

**Comment by murataya** [comment]
Thank you for your report.

Regarding your another question, I'm not sure what you want to do.
Does it mean that you want to release the listen port of smtp in a short time?
I have not implemented any options for this.

**Comment by dladeli** [comment]
Yes i need the fast release the listen port after send an email.
